src/Hosting/WindowsServices/src/WindowsServiceLifetime.cs

```diff
@@ -56,7 +62,7 @@ private void Run()
 
         public Task StopAsync(CancellationToken cancellationToken)
         {
-            Stop();
+            ApplicationLifetime.StopApplication();
```

Comment:
StopAsync is part of IHostLifetime, not ServiceBase. This is called when the app is already shutting down and we need to signal ServiceBase. You should not have needed to change this.

Reply:
I can add it back, is calling it in Task.Run ok? Otherwise it deadlocks because the host waits for StopAsync before triggering ApplicationStopped, and this method won't return until ApplicationStopped is fired.

Reply:
You want to move the Task.Run(Stop) from ApplicationStopping to here? That's OK so long as you add comments as to why.